File: monotonic-stack/next-greater-element-ii.java
class Solution {
    public int[] nextGreaterElements(int[] nums) {
        int len = nums.length;
        int[] answer = new int[len];
        Arrays.fill(answer, -1);

        for(int i = 0; i<len; i++){
            
            int index = i+1;
            for(int j = 0; j<len; j++){
                if(index==len){
                    index = 0;
                }
                
                if(nums[index]>nums[i]){
                    answer[i] = nums[index];
                    break;
                }
                index++;

            }

        }
        return answer;
    }
}

File: stack/next-greater-element-ii.py
class Solution(object):
    def nextGreaterElements(self, nums):
        result = [-1]*len(nums)
        nums = nums*2
        stack = []
        for i in range(len(nums)):
            while len(stack) > 0 and nums[stack[-1]]< nums[i]:
                index = stack.pop()
                result[index] = nums[i]
            
            stack.append(i%len(result))
        return result